src/films/hdrfilm.cpp

```diff
@@ -256,7 +256,7 @@ class HDRFilm final : public Film<Float, Spectrum> {
         ref<Bitmap> source = new Bitmap(m_channels.size() != 5 ? Bitmap::PixelFormat::MultiChannel
                                                                : Bitmap::PixelFormat::XYZAW,
                           struct_type_v<ScalarFloat>, m_storage->size(), m_storage->channel_count(),
-                          (uint8_t *) storage.data());
+                          m_channels, (uint8_t *) storage.data());
```

Comment:
Should we fix the `prepare` method above as well?

Reply:
What would need fixing there?

Reply:
Ah, you mean the sorting that's broken? Right, I had forgotten about that. I will commit it.

Reply:
THanks!